### PR TITLE
Base url note

### PR DESCRIPTION
**Pull request by JEuler**

### CODE REVIEW

File: faq.md [hunk 1]
```diff
@@ -61,6 +62,22 @@ Request compressRequest(Request request) {
 Future<Response> postRequest(@Body() Map<String, String> data);
 ```
 
+## Runtime baseUrl change
+
+Sometimes you may need to change runtime base url, for example, you have server selector and you want to use different api routes like
```
Comment (stewemetal) [inline faq.md:67]: I would rephrase this line like this:

You may need to change the base URL of your network calls during runtime, for example, if you have to use different servers or routes dynamically in your app in case of a "regular" or a "paid" user.

File: faq.md [hunk 2]
```diff
@@ -61,6 +62,22 @@ Request compressRequest(Request request) {
...
+`regular` and `pro` routes. You can store the current server base url in your SharedPreferences (encrypt/decrypt it if needed) and
+use in interceptor like that:
```
Comment (stewemetal) [inline faq.md:69]: Why is this a new line? :)

File: faq.md [hunk 3]
```diff
@@ -61,6 +62,22 @@ Request compressRequest(Request request) {
...
+
+```dart
+...
+(Request request) async =>
+              SharedPreferences.containsKey('baseUrl')
+                  ? request.copyWith(
+                      baseUrl: SharedPreferences.getString('baseUrl'))
+                  : request
+...
+```
```
Comment (stewemetal) [inline faq.md:79]: Good example, good code! 👍

File: faq.md (hunk 2 above)
Comment (JEuler) [reply faq.md:69]: Ooops, I'm dumb :D Just wrapped textline in VS Code :D

File: faq.md (hunk 1 above)
Comment (JEuler) [reply faq.md]: Thanks! Pushed your variant.

**Comment by JEuler** [comment]
@stewemetal All is okay now? :)

**Comment by JEuler** [comment]
@stewemetal ? :) Maybe @lejard-h ? :)

File: faq.md [hunk 4]
```diff
@@ -61,6 +62,20 @@ Request compressRequest(Request request) {
 Future<Response> postRequest(@Body() Map<String, String> data);
 ```
 
+## Runtime baseUrl change
+
+You may need to change the base URL of your network calls during runtime, for example, if you have to use different servers or routes dynamically in your app in case of a "regular" or a "paid" user. You can store the current server base url in your SharedPreferences (encrypt/decrypt it if needed) and use in interceptor like that:
```
[the remark this answers is not in the document]
Comment (JEuler) [reply faq.md]: Thanks, added your note! And squashed commits.
